components/raftstore/src/store/worker/region.rs

```diff
@@ -549,6 +543,78 @@ where
         }
         false
     }
+
+    fn delete_all_in_range(&self, mut ranges: Vec<(Vec<u8>, Vec<u8>)>) -> Result<()> {
+        if ranges.len() < 4 || self.use_delete_range {
+            for (start_key, end_key) in ranges {
+                for cf in self.engines.kv.cf_names() {
+                    let strategy = if cf == CF_LOCK {
+                        DeleteStrategy::DeleteByKey
+                    } else {
+                        self.get_destroy_peer_strategy()
+                    };
+                    box_try!(self
+                        .engines
+                        .kv
+                        .delete_all_in_range_cf(cf, strategy, &start_key, &end_key));
+                }
+            }
+        } else {
+            ranges.sort_by(|a, b| a.0.cmp(&b.0));
+            let mut writers = HashMap::default();
+            for cf in &[CF_DEFAULT, CF_WRITE] {
+                let sst_path = self.mgr.get_temp_path_for_ingest();
+                let builder = EK::SstWriterBuilder::new()
+                    .set_db(&self.engines.kv)
+                    .set_cf(cf);
+                let writer = box_try!(builder.build(sst_path.as_str()));
+                writers.insert(cf.to_owned(), (writer, 0, sst_path));
+            }
+
+            for (start_key, end_key) in ranges {
+                for cf in self.engines.kv.cf_names() {
+                    if let Some((writer, mut count, _)) = writers.get_mut(cf) {
+                        // We write keys of multiple regions into one sst-writer to avoid ingest too many
+                        // files into engine.
+                        box_try!(self.engines.kv.scan_cf(
```

Comment:
how about changing trait function`delete_all_in_range_cf` to `delete_ranges_cf`, so we can reuse it here